src/stepper.c

```diff
@@ -330,7 +328,6 @@ stepper_shutdown(void)
     uint8_t i;
     struct stepper *s;
     foreach_oid(i, s, command_config_stepper) {
-        s->first = NULL;
```

Comment:
The shutdown handler in basecmd.c:move_reset() always rebuilds the free list.  So, not clearing the local list here could result in the freelist being corrupted.

Reply:
This one should happen automatically:
For each *s, `stepper_stop(s)` is called. This now pops and frees every event in the queue (`while ((e = mq_event_pop(&s->queue)))` stepper.c:320).
This sets the `queue->first` to `NULL` implicitly, because the last's next-element is NULL (queue.c:24)

Reply:
Yes, but the reason for the explicit `s->first = NULL` is that we do not want to free the move_queue items in `stepper_shutdown()`.  The `move_reset()` code automatically frees all move_queue items (irrespective of owner) on a shutdown.  If we add all move_queue items to the freelist in `move_reset()` and in `stepper_shutdown()` then we'll corrupt the freelist.

Reply:
Ah, I understand.